### DISCUSSION TITLE
Possibility of doing postprocessing after finishing a simulation

### DISCUSSION THREAD

**Discussion by YvesLing** [idaholab/moose#20509] [Q&A General] [answered]
Hi Moose experts,
I am new for moose and trying to go through some tutorials. However, I am wondering if there is a way to do postprocessing after finishing a simulation rather than at the end of each iteration, for example to compute the maximum value of the solution variable. If yes, is there some introduction?

**Comment by YvesLing** [comment]
> Hi Moose experts, I am new for moose and trying to go through some tutorials. However, I am wondering if there is a way to do postprocessing after finishing a simulation rather than at the end of each iteration, for example to compute the maximum value of the solution variable. If yes, is there some introduction?

Sorry, maybe not clear, the question. What I mean is: after finising a simulation, can I do postprocessing by reading data from the output file xxx_out.e? Sometimes, it is time consuming to re-run the simulation adding a post-processing block in the input file.

**Comment by GiudGiud** [comment, marked as answer]
Hello

I usually use the restart/recover system to do this (see https://mooseframework.inl.gov/application_usage/restart_recover.html). You load the final variables from a previous solve, using either the checkpoint files or using an exodus output, then set
```
[Problem]
  solve = false
[]
```
to avoid having another solve.

Guillaume

**Reply by ABallisat** [reply]
If you are happy to do the analysis outside MOOSE you can also the SEACAS library (https://github.com/gsjaardema/seacas) which has some Python wrappers for reading EXODUS files. You can then do any arbitrary analysis on the outputted data using it, I've found it to be very useful for wrangling large numbers of output files.

Alex

**Reply by YvesLing** [reply]
I got it. Thanks.
